java/src/main/java/io/cucumber/gherkin/GherkinLine.java

```diff
@@ -70,35 +70,69 @@ boolean startsWith(String prefix) {
         return text.startsWith(prefix);
     }
 
+    public boolean startsWith(char tableCellSeparator) {
+        return !empty && text.charAt(0) == tableCellSeparator;
+    }
+
     String substringTrimmed(int beginIndex) {
-        return text.substring(beginIndex).trim();
+        // trim the beginning of the line (the end of line has already been trimmed in the constructor)
+        while ((beginIndex < textLength) && (text.charAt(beginIndex) <= ' ')) {
+            beginIndex++;
+        }
+        return text.substring(beginIndex);
     }
 
     List<GherkinLineSpan> parseTags() {
         // in most cases, the line contains no tag, so the code is optimized for this situation
         if (empty) {
             return emptyList();
         }
-        String uncommentedLine = StringUtils.removeComments(text);
-        int indexInUncommentedLine = 0;
 
-        String[] elements = uncommentedLine.split(TAG_PREFIX);
-        if (elements.length == 0) {
+        // look for the first tag
+        int indexStartTag = text.indexOf(TAG_PREFIX_CHAR);
+        int indexComment = text.indexOf(COMMENT_PREFIX_CHAR);
+        if (indexStartTag < 0 || (indexComment >= 0 && indexStartTag > indexComment)) {
+            // no tag found (or all tags are commented out)
             return emptyList();
         }
-        List<GherkinLineSpan> tags = new ArrayList<>(elements.length);
-        for (String element : elements) {
-            String token = rtrim(element);
-            if (token.isEmpty()) {
-                continue;
+
+        List<GherkinLineSpan> tags = new ArrayList<>();
+        int indexEndOfLine = indexComment >= 0 ? indexComment : text.length();
```

Comment:
Oops, I didn't know that a tag can contain a '#' (e.g. "@comment_tag#2" like in `tags.feature`). This is why the "make acceptance" fails. I'll provide a correction.

Reply:
I saw comments that start at the beginning of the line (so without whitespace before the '#'), e.g. in `descriptions_with_comments.feature:16`:

    Scenario: without indentation
    This is a description without indentation
    # comment
    and a comment in the middle and at the end
    # comment 2

Is the `<WHITESPACE_CHARS>#` only for lignes that contains tags ?
Or is the true comment expression `^#|<WHITESPACE_CHARS>#` ?

Reply:
No, only for lines with a tag. I.e. lines that start with `@` after an left trim.

Reply:
Lines with a tag are also the only lines that allow trailing comments. On every other line it would be part of the step, title, description, ect.